src/openfl/display/Stage.hx

```diff
@@ -1916,7 +1916,16 @@ class Stage extends DisplayObjectContainer implements IModule {
 	
 	private function __startDrag (sprite:Sprite, lockCenter:Bool, bounds:Rectangle):Void {
 		
-		__dragBounds = (bounds == null) ? null : bounds.clone ();
+		if (bounds == null) {
+			__dragBounds = null;
+		} else {
+			__dragBounds = new Rectangle ();
+			__dragBounds.x = bounds.right < bounds.x ? bounds.right : bounds.x;
```

Comment:
tiny nitpick: `right` and `bottom` are actually getter functions, so maybe it's better to cache the values